peerdid/peer_did_utils.py

```diff
@@ -0,0 +1,232 @@
+    encnumbasis = base58.b58encode(prefixed_decoded_key)
+    return encnumbasis.decode('utf-8')
+
+
+def _decode_encnumbasis(encnumbasis: str, peer_did: PEER_DID) -> dict:
+    """
+    Decodes encnumbasis
+    :param encnumbasis: encnumbasis to decode
+    :param peer_did: peer_did which will be used as an ID
+    :return: decoded encnumbasis
+    """
+    decoded_encnumbasis = base58.b58decode(encnumbasis)
+    codec = _get_codec(data=bytes(decoded_encnumbasis))
+    decoded_encnumbasis_without_prefix = _remove_prefix(decoded_encnumbasis)
+    public_key = base58.b58encode(decoded_encnumbasis_without_prefix).decode('utf-8')
+    return {'id': peer_did + '#' + encnumbasis, 'type': codec, 'controller': peer_did, 'publicKeyBase58': public_key}
+
+
+def _remove_prefix(data: bytes) -> bytes:
+    """
+    Removes prefix from data
+    :param data: prefixed data
+    :return: data without prefix
+    """
+    prefix_int = _extract_prefix(data)
+    prefix = varint.encode(prefix_int)
+    return data[len(prefix):]
+
+
+def _get_codec(data: bytes) -> str:
+    """
+    Gets codec from data
+    :param data: prefixed data
+    :raises ValueError: if prefix is not supported
+    :return: codec name
+    """
+    prefix = _extract_prefix(data)
+    if prefix in set(item.value for item in PublicKeyTypeAuthentication):
+        return PublicKeyTypeAuthentication(prefix).name
+    elif prefix in set(item.value for item in PublicKeyTypeAgreement):
+        return PublicKeyTypeAgreement(prefix).name
+    else:
+        raise ValueError('Prefix {} not present in the lookup table'.format(prefix))
+
+
+def _extract_prefix(data: bytes) -> int:
+    """
+    Extracts prefix from data
+    :param data: prefixed data
+    :raises ValueError: if invalid varint provided
+    :return: prefix
+    """
+    try:
+        return varint.decode_bytes(data)
+    except TypeError:
+        raise ValueError('incorrect varint provided')
+
+
+def _add_prefix(key_type: Union[PublicKeyTypeAgreement, PublicKeyTypeAuthentication], data: bytes) -> bytes:
+    """
+    Adds prefix to a data
+    :param key_type: type of key
+    :param data: data to be prefixed
+    :return: prefixed data
+    """
+    prefix = varint.encode(key_type.value)
+    return b''.join([prefix, data])
+
+
+def _encode_filename(filename: str) -> str:
+    """
+    Encodes filename to SHA256 string
+    :param filename: name of file
+    :return: encoded filename as SHA256 string
+    """
+    return hashlib.sha256(filename.encode()).hexdigest()
+
+
+def _build_did_doc_numalgo_0(peer_did: PEER_DID) -> dict:
+    """
+    Helper method to create did_doc according to numalgo 0
+    :param peer_did: peer_did to resolve
+    :raises ValueError:
+    1) if peer_did contains encryption key instead of signing
+    2) if peer_did contains unsupported transform part
+    :return: did_doc
+    """
+    inception_key = peer_did[11:]
+    transform = peer_did[10]
+    if not transform == 'z':
```

Comment:
We should have this check in `_build_did_doc_numalgo_2` keys as well. Probably let's move it to `_decode_encnumbasis`?